src/trade/BarrierInput.js

```diff
@@ -0,0 +1,22 @@
+import React, { PropTypes, Component } from 'react';
+import { InputGroup } from '../_common';
+
+export default class BarrierInput extends Component {
+    static propTypes = {
+        name: PropTypes.string.isRequired,
+        onChange: PropTypes.func,
+        defaultValue: PropTypes.number,
+    };
```

Comment:
I started listing the props ordered alphabetically.
Try it next time, I am also not following that very actively.